MediaPlayerPrivateMediaSourceAVFObjC is not handling aspect ratio preservation signal from HTMLMediaElement

youennf opened the pull request:
#### a8b3f78d322baecab6e30c5e454410a25f542740
<pre>
MediaPlayerPrivateMediaSourceAVFObjC is not handling aspect ratio preservation signal from HTMLMediaElement
<a href="https://bugs.webkit.org/show_bug.cgi?id=264944">https://bugs.webkit.org/show_bug.cgi?id=264944</a>
<a href="https://rdar.apple.com/118508061">rdar://118508061</a>

Reviewed by Eric Carlson and Jer Noble.

MediaPlayer is calling setShouldMaintainAspectRatio but MediaPlayerPrivateMediaSourceAVFObjC was not implementing it.
Implement it by setting the gravity to AVLayerVideoGravityResizeAspect when we should maintain aspectRatio and AVLayerVideoGravityResizeAspect otherwise.
This fixes the bug for UI side compositing.
We should do a follow-up to fix it for non UI side compositing.

* LayoutTests/http/tests/media/media-source/media-source-video-fit-fill-expected.txt: Added.
* LayoutTests/http/tests/media/media-source/media-source-video-fit-fill.html: Added.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::setShouldMaintainAspectRatio):

Canonical link: <a href="https://commits.webkit.org/270876@main">https://commits.webkit.org/270876@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f09fe569d370c6ea56bb9164f579a2dc7aef86d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26705 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5319 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27945 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28917 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24416 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/27150 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7143 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2716 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/24327 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26967 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4135 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22924 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3635 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3674 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23916 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29402 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24349 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24316 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29940 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3712 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1908 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27839 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5161 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/23652 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6408 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4181 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4065 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jernoble commented on Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
```diff
@@ -1557,6 +1558,21 @@ void getSupportedTypes(HashSet<String>& types) const final
+{
+    m_shouldMaintainAspectRatio = shouldMaintainAspectRatio;
+
+    if (m_sampleBufferDisplayLayer) {
```
I'd bail out early if `m_shouldMaintainAspectRatio == shouldMaintainAspectRatio `, and early return if `! m_sampleBufferDisplayLayer`.

youennf replied:
ok

youennf committed:
MediaPlayerPrivateMediaSourceAVFObjC is not handling aspect ratio preservation signal from HTMLMediaElement
https://bugs.webkit.org/show_bug.cgi?id=264944
rdar://118508061

Reviewed by Eric Carlson and Jer Noble.

MediaPlayer is calling setShouldMaintainAspectRatio but MediaPlayerPrivateMediaSourceAVFObjC was not implementing it.
Implement it by setting the gravity to AVLayerVideoGravityResizeAspect when we should maintain aspectRatio and AVLayerVideoGravityResizeAspect otherwise.
This fixes the bug for UI side compositing.
We should do a follow-up to fix it for non UI side compositing.

* LayoutTests/http/tests/media/media-source/media-source-video-fit-fill-expected.txt: Added.
* LayoutTests/http/tests/media/media-source/media-source-video-fit-fill.html: Added.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::setShouldMaintainAspectRatio):

Canonical link: https://commits.webkit.org/270876@main
